browser/ui/webui/speedreader/speedreader_toolbar_ui.cc

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ui/webui/speedreader/speedreader_toolbar_ui.h"
+
+#include <utility>
+
+#include "brave/browser/ui/webui/brave_webui_source.h"
+#include "brave/components/ai_chat/common/features.h"
+#include "brave/components/constants/webui_url_constants.h"
+#include "brave/components/l10n/common/localization_util.h"
+#include "brave/components/speedreader/common/constants.h"
+#include "brave/components/speedreader/resources/panel/grit/brave_speedreader_toolbar_generated_map.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/browser_finder.h"
+#include "components/grit/brave_components_resources.h"
+#include "content/public/browser/host_zoom_map.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/browser/web_ui_data_source.h"
+#include "content/public/common/url_constants.h"
+
+SpeedreaderToolbarUI::SpeedreaderToolbarUI(content::WebUI* web_ui,
+                                           const std::string& name)
+    : ui::MojoBubbleWebUIController(web_ui, true),
+      profile_(Profile::FromWebUI(web_ui)) {
+  content::HostZoomMap::Get(web_ui->GetWebContents()->GetSiteInstance())
+      ->SetZoomLevelForHostAndScheme(content::kChromeUIScheme,
+                                     kSpeedreaderPanelHost, 0);
+
+  browser_ = chrome::FindLastActiveWithProfile(profile_);
+
+  content::WebUIDataSource* source = CreateAndAddWebUIDataSource(
+      web_ui, name, kBraveSpeedreaderToolbarGenerated,
+      kBraveSpeedreaderToolbarGeneratedSize, IDR_SPEEDREADER_UI_HTML);
+
+  for (const auto& str : speedreader::kLocalizedStrings) {
+    std::u16string l10n_str =
+        brave_l10n::GetLocalizedResourceUTF16String(str.id);
+    source->AddString(str.name, l10n_str);
+  }
+
+  source->AddBoolean("aiChatFeatureEnabled",
+                     ai_chat::features::IsAIChatEnabled());
```

Comment:
This needs a buildflag wrapper, and provided `false` if the buildflag is disabled